heroic-core/src/main/java/com/spotify/heroic/shell/task/AnnotatedExport.java

```diff
@@ -0,0 +1,260 @@
+public class AnnotatedExport implements ShellTask {
+    private final AsyncFramework async;
+    private final Clock clock;
+    private final MetadataManager metadata;
+    private final MetricManager metrics;
+    private final QueryParser parser;
+    private final ObjectMapper mapper;
+
+    @Inject
+    public AnnotatedExport(
+        AsyncFramework async, Clock clock, MetadataManager metadata, MetricManager metrics,
+        QueryParser parser, @Named("application/json") ObjectMapper mapper
+    ) {
+        this.async = async;
+        this.clock = clock;
+        this.metadata = metadata;
+        this.metrics = metrics;
+        this.parser = parser;
+        this.mapper = mapper;
+    }
+
+    @Override
+    public TaskParameters params() {
+        return new Parameters();
+    }
+
+    @Override
+    public AsyncFuture<Void> run(final ShellIO io, final TaskParameters base) throws Exception {
+        final Parameters params = (Parameters) base;
+
+        final Filter filter = Tasks.setupFilter(parser, params);
+
+        final MetadataBackend group = metadata.useDefaultGroup();
+        final MetricBackendGroup m = metrics.useDefaultGroup();
+
+        final ResolvableFuture<Void> future = async.future();
+
+        final long now = clock.currentTimeMillis();
+
+        final long start = params.start.map(t -> Tasks.parseInstant(t, now)).orElse(now);
+        final long end = params.end
+            .map(t -> Tasks.parseInstant(t, now))
+            .orElseThrow(() -> new RuntimeException("end: is required"));
+
+        final DateRange range = new DateRange(start, end);
+
+        group
+            .findSeriesStream(new FindSeries.Request(filter, params.getRange(), params.getLimit()))
+            .observe(new AsyncObserver<FindSeriesStream>() {
+                final ConcurrentLinkedQueue<Series> allSeries = new ConcurrentLinkedQueue<>();
+
+                @Override
+                public AsyncFuture<Void> observe(final FindSeriesStream value) {
+                    allSeries.addAll(value.getSeries());
+                    return async.resolved();
+                }
+
+                @Override
+                public void cancel() {
+                    future.cancel();
+                }
+
+                @Override
+                public void fail(final Throwable cause) {
+                    future.fail(cause);
+                }
+
+                @Override
+                public void end() {
+                    final List<Callable<AsyncFuture<SeriesAndData>>> callables = new ArrayList<>();
+
+                    for (final Series s : allSeries) {
+                        callables.add(() -> {
+                            final FetchData.Request request =
+                                new FetchData.Request(MetricType.POINT, s, range,
+                                    QueryOptions.defaults());
+                            return m
+                                .fetch(request, FetchQuotaWatcher.NO_QUOTA)
```

Comment:
This uses the deprecated variant of fetch(), so non-streaming. As long as the data fits in memory that's not a big problem, but if we want to export more then it would have been great to do it in a streaming way. How about adding a TODO or fix?